### PR TITLE
DOC-1602: Added documentation for the new model files/options

### PR DESCRIPTION
**Pull request by TheSpyder**
Related Ticket: DOC-1602

Description of Changes:
* Added `Content -> Editor content models` (structure based on `Interface -> Appearance -> Themes`)
* Updated `How-to guides -> Bundling TinyMCE -> *` to include model as a peer of theme in every appropriate place
* Fixed a few inconsistencies in language used by the various bundle pages
* Added references to model in the only places it seemed appropriate (baseurl and folder lists)

This deliberately does not document `editor.model` or `tinymce.ModelManager` as they are not intended for use by customers. Also note the wording I used in `pages/editor-model.adoc` is my own but I have sent it to @lostkeys for approval.

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] ~Files removed have been deleted, not just excluded from the build (if applicable)~
- [x] ~(New product features only) Release Note added~

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: modules/ROOT/pages/bundling-models.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
+= Bundling TinyMCE models using module loading
+
+:title_nav: Models
+:description_short: Information on bundling Models
```
Comment (lnewson) [inline modules/ROOT/pages/bundling-models.adoc:4]:
```suggestion
:description_short: Information on bundling models
```

File: modules/ROOT/pages/bundling-skins.adoc [hunk 2]
```diff
@@ -34,31 +34,31 @@ IMPORTANT: The handling of content CSS files (such as `+content.css+` or `+conte
 
 [source,js]
 ----
-import contentUiCss from 'tinymce/skins/ui/example/content.css';
+import contentUiSkin from 'tinymce/skins/ui/example/content.css';
```
Comment (lnewson) [inline modules/ROOT/pages/bundling-skins.adoc:37]: Why was this changed? This isn't related to models and doesn't seem correct as it is a CSS file... if it is then the inline variable below should also be changed.

- [ ] Revert these changes please

File: modules/ROOT/partials/configuration/model.adoc [hunk 3]
```diff
@@ -0,0 +1,20 @@
+[[model]]
+== `+model+`
+
+This option allows you to specify the model that {productname} should use. The default model included with {productname} is *dom*. Custom models are not supported at this time.
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/model.adoc:4]: `you` should be avoided... a lot of old docs use it still but new docs shouldn't. See https://www.tiny.cloud/docs/configure/contributing-docs/#pronouns

```suggestion
This option allows configuring which model {productname} should use. The default model included with {productname} is *dom*. Custom models are not supported at this time.
```

File: modules/ROOT/partials/configuration/model_url.adoc [hunk 4]
```diff
@@ -0,0 +1,16 @@
+[[model_url]]
+== `+model_url+`
+
+This option enables you to specify the location of the `dom` model file. This is useful if you are loading {productname} from one URL, for example a CDN, while loading a model on, say, a local server.
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/model_url.adoc:4]:
```suggestion
This option allows specifying the location of the `dom` model file. This is useful if you are loading {productname} from one URL, for example a CDN, while loading a model on, say, a local server.
```
Again, we shouldn't be using `you`.

File: modules/ROOT/partials/configuration/model.adoc [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+The name of the model should match the name of the folder within the models directory of {productname}. If the specified model is not found, {productname} will not load.
+
+Type : `+String+`
+
+Default Value : `dom`
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/model.adoc:10]:
```suggestion
Default Value : `'dom'`
```

File: modules/ROOT/partials/configuration/theme.adoc [hunk 6]
```diff
@@ -7,6 +7,8 @@ The name of the theme should match the name of the folder within the themes dire
 
 Type : `+String+`
 
+Default Value : `silver`
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/theme.adoc:10]:
```suggestion
Default Value : `'silver'`
```

File: modules/ROOT/pages/editor-model.adoc [hunk 7]
```diff
@@ -0,0 +1,10 @@
+= TinyMCE Model options
+
+:title_nav: Models
+:description: Configure the editor's content model.
+
+WARNING: {productname} model configuration is a future feature. The only supported models are the default (`+dom+`) and the one provided by the the premium https://www.tiny.cloud/tinymce/features/real-time-collaboration/[real-time collaboration] feature. These options exist only for use in scenarios where the `dom` model cannot load automatically.
```
Comment (lnewson) [inline modules/ROOT/pages/editor-model.adoc:6]:
```suggestion
WARNING: {productname} model configuration is a future feature. The only supported models are the default (`+dom+`) and the one provided by the the premium https://www.tiny.cloud/tinymce/features/real-time-collaboration/[Real-time Collaboration] feature. These options exist only for use in scenarios where the `dom` model cannot load automatically.
```
This is a name, so should match the name on the page.

File: modules/ROOT/pages/editor-model.adoc [hunk 8]
```diff
@@ -0,0 +1,10 @@
+= TinyMCE Model options
+
+:title_nav: Models
```
Comment (lnewson) [inline modules/ROOT/pages/editor-model.adoc:3]: This is technically a useless attribute atm and I'd like to remove it, but for now I guess it should match the actual name you've used in the nav.
```suggestion
:title_nav: Editor content models
```

**Review by lnewson — APPROVED** [review]

File: modules/ROOT/pages/bundling-skins.adoc (hunk 2 above)
Comment (lnewson) [reply modules/ROOT/pages/bundling-skins.adoc:37]: I should add that `Ui` and `skin` both mean the same thing here so this new name basically doubles up. At the very least, if you don't want to revert this should still include `css` in the name since that's what it's holding.

Comment (TheSpyder) [reply modules/ROOT/pages/bundling-skins.adoc:37]: I was already changing this nearly-identical file a bunch of times and noticed they weren't actually identical. At least when they're properly copy/paste they can be updated easily with a search and replace.

The folder location includes both ui and skin, so I figured the variable name could too 🤷 the point I was trying to hint at to customers is part of the skin which isn't optional, whereas regular content CSS is. I guess I'll make it `contentUiSkinCss`.

File: modules/ROOT/partials/configuration/model.adoc (hunk 3 above)
Comment (TheSpyder) [reply modules/ROOT/partials/configuration/model.adoc:4]: ok, this whole thing is 90% the `theme.adoc` file with `theme` replaced by `model`.

File: modules/ROOT/partials/configuration/model_url.adoc (hunk 4 above)
Comment (TheSpyder) [reply modules/ROOT/partials/configuration/model_url.adoc:4]: Ditto from `model.adoc`.

File: modules/ROOT/partials/configuration/model.adoc (hunk 3 above)
Comment (lnewson) [reply modules/ROOT/partials/configuration/model.adoc]: Yeah I figured that was probably the case.

File: modules/ROOT/pages/bundling-skins.adoc (hunk 2 above)
Comment (lnewson) [reply modules/ROOT/pages/bundling-skins.adoc:37]: Fair point about the `ui/skin` path, so `contentUiSkinCss` sounds fine.

Re the differences, looks like when that was done someone accidentally swapped the order previously.

File: modules/ROOT/partials/configuration/model_url.adoc [hunk 9]
```diff
@@ -0,0 +1,16 @@
+[[model_url]]
+== `+model_url+`
+
+This option allows specifying the location of the `dom` model file. This is useful if you are loading {productname} from one URL, for example a CDN, while loading a model on, say, a local server.
```
Comment (ltrouton) [inline modules/ROOT/partials/configuration/model_url.adoc:4]:
```suggestion
This option allows specifying the location of the `dom` model file. This is useful if loading {productname} from one URL, for example, a CDN, while loading a model on, say, a local server.
```

File: modules/ROOT/partials/module-loading/bundling-rollup-es6-npm_editor.adoc [hunk 10]
```diff
@@ -9,13 +9,14 @@ Example `+src/editor.js+`
...
+/* Default icons are required. After that, import custom icons if applicable */
 import 'tinymce/icons/default';
 
-/* A theme is also required */
+/* Required TinyMCE components */
 import 'tinymce/themes/silver';
+import 'tinymce/models/dom';
 
-/* Import the skin */
+/* Import a skin (can be your own instead of the default) */
```
Comment (ltrouton) [inline modules/ROOT/partials/module-loading/bundling-rollup-es6-npm_editor.adoc:19]: Should we try and avoid saying 'your own...' in the comment here and in the other similar files as well?
Possible alternative?
```suggestion
/* Import a skin (can be a custom skin instead of the default) */
```

**Review by ltrouton — APPROVED** [review]

**Review by EkimChau — APPROVED** [review]

File: modules/ROOT/partials/module-loading/bundling-rollup-es6-npm_editor.adoc (hunk 10 above)
Comment (TheSpyder) [reply modules/ROOT/partials/module-loading/bundling-rollup-es6-npm_editor.adoc:19]: ah yeah we are avoiding that language aren't we. I'll do a pass over it.
